Firebase/Messaging/FIRMessaging.m

```diff
@@ -413,13 +413,18 @@ - (void)handleIncomingLinkIfNeededFromMessage:(NSDictionary *)message {
   // Similarly, |application:openURL:sourceApplication:annotation:| will also always be called, due
   // to the default swizzling done by FIRAAppDelegateProxy in Firebase Analytics
   } else if ([appDelegate respondsToSelector:openURLWithSourceApplicationSelector]) {
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
     [appDelegate application:application
                      openURL:url
            sourceApplication:FIRMessagingAppIdentifier()
                   annotation:@{}];
-
+#pragma clang diagnostic pop
   } else if ([appDelegate respondsToSelector:handleOpenURLSelector]) {
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
```

Comment:
nit: You can probably just wrap both `else` statements in a single `#pragma`, since this code is unlikely to change.

Reply:
Just reset my laptop and lost this client, will redo in another CL.